enterprise/static/enterprise/js/manage_learners.js

```diff
@@ -134,11 +140,26 @@ function loadPage() {
        programEnrollment.$control.oldValue = null;
     });
 
+    $("#id_bulk_upload_csv").change(function(e) {
+        if (e.target.value) {
+            var force_enrollment = $("#id_force_enrollment");
+            force_enrollment.parent().show();
+            force_enrollment.siblings(".helptext")[0].innerHTML = gettext(
+                "If any of the courses in the CSV file are marked 'Invite Only', " +
+                "this should be enabled for the enrollments to go through in those courses."
+            );
+        }
+    });
```

Comment:
This part is has a potential side effect of getting learners enrolled into courses which might be closed for reasons other than being marked "Invite Only". Suggestions for possible solutions for this side-effect are welcome.

Reply:
@tecoholic, this sounds like a reasonable tradeoff.

A proper solution I can think of would be adding more checks around [this code](https://github.com/openedx/edx-platform/blob/065ba5fd7360b20756dce81be0449976a0fb596e/common/djangoapps/student/models/course_enrollment.py#L690-L696) to verify if the requester is `staff` (e.g., with `crum.get_current_user`) and if the course is `invitation_only`.

Currently, the [EnrollmentListView](https://github.com/openedx/edx-platform/blob/065ba5fd7360b20756dce81be0449976a0fb596e/openedx/core/djangoapps/enrollments/views.py#L446-L446) does not support invitation only courses. This problem does not happen in the Instructor Dashboard because it doesn't invoke access checks [when calling the `enroll` method](https://github.com/openedx/edx-platform/blob/065ba5fd7360b20756dce81be0449976a0fb596e/common/djangoapps/student/models/course_enrollment.py#L772-L772). Modifying the `enroll`  method is a high-risk change, and it seems that the API will need some changes in the near future anyway - when the Instructor Dashboard is converted into an MFE (it could be an excellent opportunity to eliminate many legacy behaviors there, too). Therefore, the solution from this PR is "good enough" at the moment.

Please add more context as a comment in the code to explain why we are introducing this functionality. Feel free to use this explanation if you'd like.

Reply:
Thanks for the detailed explanation. I tried to work as much as possible using the existing platform APIs and didn't want to introduce any changes on the platform, as I wasn't sure about the wider implications it might have. I am glad you found it a reasonable tradeoff as well. 

I have added your explanation as a part of the comment in the code. :)